src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -728,6 +728,9 @@ public ImmutableArray<T> Replace(T oldValue, T newValue)
         public ImmutableArray<T> Replace(T oldValue, T newValue, IEqualityComparer<T> equalityComparer)
         {
             var self = this;
+            // TODO: This is calling the IImmutableList<T> overload which is silently boxing the ImmutableArray<T>.
+            // It's also causing inconsistencies where calling Replace on an IsDefault instance will throw an
+            // InvalidOperationException rather than a NullReferenceException.
```

Comment:
@AArnott OK, I have reverted the comment and opened https://github.com/dotnet/corefx/issues/14961 to be addresed after the test changes go in.